src/plugins/telemetry_collection_manager/server/plugin.ts

```diff
@@ -292,17 +306,39 @@ export class TelemetryCollectionManagerPlugin
       logger: this.logger.get(collection.title),
       version: this.version,
     };
-
     const clustersDetails = await collection.clusterDetailsGetter(statsCollectionConfig, context);
+    const { refreshCache } = statsCollectionConfig;
+    const { title: collectionSource } = collection;
+    this.cacheManager.unrefExpiredCacheObjects();
+
+    // on `refreshCache: true` clear all cache to store a fresh copy
+    if (refreshCache) {
+      this.cacheManager.unrefAllCacheObjects();
+    }
 
     if (clustersDetails.length === 0) {
-      // don't bother doing a further lookup.
       return [];
     }
 
+    const cacheKey = this.createCacheKey(collection, clustersDetails);
+    if (this.cacheManager.isCacheValid(cacheKey)) {
+      const cachedData = this.cacheManager.getFromCache<BasicStatsPayload[]>(cacheKey);
+      if (cachedData) {
+        const { data, cacheTimestamp } = cachedData;
+        return data.map((stat) => ({
+          cacheDetails: {
+            isCached: true,
+            cacheTimestamp,
```

Comment:
why do cache details leak outside?

Reply:
I figured we'd want to capture this information in the usage payload.